lib/ansible/modules/cloud/azure/azure_rm_sql_server.py

```diff
@@ -0,0 +1,292 @@
+        required: False
+    version:
+        description:
+            - The version of the server.
+        required: False
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+      - name: Create (or update) Servers
+        azure_rm_sql_server:
+          resource_group: "{{ resource_group }}"
+          name: zims-server
+          location: westus
+          tags: "{{ tags }}"
+          identity:
+            type: "{{ type }}"
+          administrator_login: mylogin
+          administrator_login_password: Testpasswordxyz12!
+          version: "{{ version }}"
+'''
+
+RETURN = '''
+state:
+    description: Current state of Server
+    returned: always
+    type: dict
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.mgmt.sql import SqlManagementClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMServers(AzureRMModuleBase):
+    """Configuration class for an Azure RM Server resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str',
+                required=True
+            ),
+            tags=dict(
+                type='dict',
+                required=False
+            ),
+            identity=dict(
+                type='dict',
+                required=False
+            ),
+            administrator_login=dict(
+                type='str',
+                required=False
+            ),
+            administrator_login_password=dict(
+                type='str',
+                required=False
+            ),
+            version=dict(
+                type='str',
+                required=False
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.parameters = dict()
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+        self.state = None
+
+        super(AzureRMServers, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                             supports_check_mode=True,
+                                             supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
```

Comment:
Have a look at [azure_rm_virtualmachine](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py) to see how to do this, you usually set the parameter as None in the __init__ and these are populated in `AzureRMModuleBase` for you.